src/basic/namespace-util.c

```diff
@@ -210,14 +213,40 @@ int userns_acquire(const char *uid_map, const char *gid_map) {
 
         /* Forks off a process in a new userns, configures the specified uidmap/gidmap, acquires an fd to it,
          * and then kills the process again. This way we have a userns fd that is not bound to any
-         * process. We can use that for file system mounts and similar. */
+         * process. We can use that for file system mounts and similar.
+         * If the owner of the new namespace is specified, setuid to it in the child. Use a pipe to sync and
+         * ensure the mapping is done after setuid(). */
```

Comment:
That comment is wrong. The mapping isn't relevant. What's relevant is that the userns is created as the user that's supposed to own it.

Reply:
doesn't the mapping have to be set up _after_ the setuid?

Reply:
No, the mapping can be written whenever the userns has been created. As soon as a process has called unshare() or forked into a new userns you can right away write the mapping.

Reply:
right, the order has to be setuid -> unshare -> map, got confused and forgot to document the middle part